prac_07/programming_language.py

```diff
@@ -1,6 +1,6 @@
 """
 CP1404/CP5632 Practical - Suggested Solution
-Programming Language class with tests.
+Programming Language class with tests..
 """
 
 
```

Comment:
```
    def __init__(self, name, typing, reflection, year, arithmetic):
        """Construct a ProgrammingLanguage from the given values."""
        self.name = name
        self.typing = typing
        self.reflection = reflection
        self.year = year
        self.arithmetic = arithmetic  # Pointer Arithmetic
  ```
  
In the case of the arithmetic attribute, if you need to provide a comment to explain the attribute name, I think its probably just a bad name.  Why not just call it:
```
self.pointer_arithmetic = pointer_arithmetic  
```

Otherwise all the attribute names here are solid and make sense, but I would consider taking it a step further and providing slightly more detail to enhance readability and provide clearer context:  
```
    def __init__(self, name="", typing="", has_reflection=False, year=0, pointer_arithmetic=False):
        """Construct a ProgrammingLanguage from the given values."""
        self.name = name
        self.typing = typing
        self.has_reflection = has_reflection
        self.year = year
        self.pointer_arithmetic = pointer_arithmetic
```